src/actors/FilterBarActor.js

```diff
@@ -89,15 +102,34 @@ export class FilterBarActor {
     SearchClient.saveSearch(
       this.filterBarStore.getSavedSearchesUrl(),
       savedSearchPacket,
-      function() {
-        SearchClient.getSavedSearches(
-          this.filterBarStore.getSavedSearchesUrl(),
-          this.filterBarStore.setSavedSearches.bind(this.filterBarStore)
-        );
-      }.bind(this)
+      this.reloadSavedSearches.bind(this)
     );
 
     this.applyFilters();
   }
+
+  deleteSavedSearch(searchId) {
+    var savedSearch = this.filterBarStore.getSavedSearch(searchId);
+
+    if (!savedSearch.url) {
+      return;
+    }
+
+    var confirmation = confirm('Unfortunately one of the filters cannot be applied anymore. Remove the saved search?');
```

Comment:
may be change this message to be...

"One of the filters in this saved search cannot be applied anymore. Remove saved search?"

I don't think we need to mention the "unfortunately" lol